File: Rust/leetcode/2-keys-keyboard/solution.rs
impl Solution {
    pub fn min_steps(n: i32) -> i32 {
        let mut n = n;
        let mut res = 0;
        let mut d = 2;
        while d <= n {
            while n % d == 0 {
                res += d;
                n /= d;
            }
            d += 1;
        }
        res
    }
}

File: CSharp/leetcode/2-keys-keyboard/solution.cs
public class Solution {
    public int MinSteps(int n) {
        int res = 0;
        for (int d = 2; d <= n; d++) {
            while (n % d == 0) {
                res += d;
                n /= d;
            }
        }
        return res;
    }
}
